src/index.ts

```diff
@@ -65,39 +83,49 @@ function extractSubgraphUrl(artifact: Artifact) {
 
 async function loadSubgraphSchemas(): Promise<ServiceDefinition[]> {
 
-    Logger.info("Fetching artifacts from Schema Registry")
-    const artifactsRespose: Response<ArtifactsResponse> = await fetchArtifacts();
-
-    const serviceDefinitions: ServiceDefinition[] = [];
-    const artifacts: Artifact[] = artifactsRespose.body.artifacts;
-
-    if (artifacts.length == 0) {
-        Logger.warn("No artifacts found at the Schame Registry")
-        return [];
-    }
-
-    Logger.info(`Found ${artifacts.length} artifacts`)
-    for (const artifact of artifacts) {
-        Logger.info(`Fetching details of artifact: ${artifact.id}`)
-        const gqlResponse: Response<string> = await fetchArtifactsDetails(artifact);
-
-        let url = extractSubgraphUrl(artifact);
-
-        serviceDefinitions.push({
-            name: artifact.id,
-            url: url,
-            typeDefs: parse(gqlResponse.body)
-        });
+    try {
+        Logger.info("Fetching artifacts from Schema Registry")
+        const artifactsRespose: Response<ArtifactsResponse> = await fetchArtifacts();
+        const serviceDefinitions: ServiceDefinition[] = [];
+        const artifacts: Artifact[] = artifactsRespose.body.artifacts;
+    
+        if (artifacts.length == 0) {
+            Logger.warn("No artifacts found at the Schema Registry")
+            return [];
+        }
+    
+        Logger.info(`Found ${artifacts.length} artifacts`)
+        for (const artifact of artifacts) {
+            try {
+                Logger.info(`Fetching details of artifact: ${artifact.id}`)
+                const gqlResponse: Response<string> = await fetchArtifactsDetails(artifact);
+
+                let url = extractSubgraphUrl(artifact);
+
+                serviceDefinitions.push({
+                    name: artifact.id,
+                    url: url,
+                    typeDefs: parse(gqlResponse.body)
+                });
+            } catch (err : any) {
+                Logger.error(`Failed to fetch artifact details with error: ${err.message}`)
+                throw(err)
+            }
+        }
+    
+        return serviceDefinitions;
+    } catch (err : any) {
```

Comment:
This try-catch block is too large, and it has an inner catch within the for-loop. Mind to restructure it a bit?

Reply:
Made some changes!

Thanks for the advise